gobblin-service/src/main/java/org/apache/gobblin/service/modules/orchestration/DagManager.java

```diff
@@ -372,16 +372,17 @@ public synchronized void setActive(boolean active) {
         //Initializing state store for persisting Dags.
         this.dagStateStore = createDagStateStore(config, topologySpecMap);
         this.failedDagStateStore = createDagStateStore(ConfigUtils.getConfigOrEmpty(config, FAILED_DAG_STATESTORE_PREFIX).withFallback(config), topologySpecMap);
+        Set<String> failedDagIds = Collections.synchronizedSet(this.failedDagStateStore.getDagIds());
```

Comment:
Can we not look up a dag with a given dagId from the dagStateStore when a resume is called? (which should be infrequent), instead of loading dagIds on start up?

Reply:
That is what I am doing, see the changes in `beginResumingDag` just below this. The reason for still keeping this list is that it's used to know which dags are old enough to delete from the failed dag state store, and also it could avoid some calls to mysql (if calling resume with an invalid dag). The list of dagIds is only strings so it shouldn't be large in memory.